backend/cmm_helpers.ml

```diff
@@ -722,16 +720,22 @@ let rec div_int ?dividend_cannot_be_min_int c1 c2 dbg =
          t = shift-right-signed(t, p)
 
          res = t + sign-bit(c1) *)
```

Comment:
you should update the algorithm to:
- match the new names below (particular the renamings such as `t -> q`, `c1 -> n`)
- make the algorithm reflect what is actually computed now as there are a few new cases (related to the handling of negative dividends if I'm not mistaken)